gtk3-macros/src/lib.rs

```diff
@@ -21,38 +21,57 @@ use syn::{parse_macro_input, DeriveInput};
 /// Specify that `MyWidget` is using a composite template and load the
 /// template file the `composite_template.ui` file.
 ///
-/// ```compile_fail
-/// use gtk::prelude::*;
-/// use gtk::CompositeTemplate;
-///
-/// #[derive(Debug, Default, CompositeTemplate)]
-/// #[template(file = "composite_template.ui")]
-/// struct MyWidget {
-///     #[template_child]
-///     pub label: TemplateChild<gtk::Label>,
-/// }
-/// ```
-///
 /// Then, in the `ObjectSubclass` implementation you will need to call
 /// `bind_template` in the `class_init` function, and `init_template` in
 /// `instance_init` function.
 ///
+/// ```no_run
+/// use gtk::prelude::*;
+/// use gtk::glib;
+/// use gtk::subclass::prelude::*;
 ///
-/// ```compile_fail
-/// #[glib::object_subclass]
-/// impl ObjectSubclass for MyWidget {
-///        const NAME: &'static str = "MyWidget";
-///        type Type = super::MyWidget;
-///        type ParentType = gtk::Widget;
-///
-///        fn class_init(klass: &mut Self::Class) {
-///            Self::bind_template(klass);
-///        }
-///
-///        fn instance_init(obj: &glib::subclass::InitializingObject<Self>) {
-///            obj.init_template();
-///        }
-///    }
+/// mod imp {
+///     use super::*;
+/// 
+///     #[derive(Debug, Default, CompositeTemplate)]
+///     #[template(file = "composite_template.ui")]
+///     pub struct MyWidget {
+///         #[template_child]
+///         pub label: TemplateChild<gtk::Label>,
```

Comment:
```suggestion
///         pub label: gtk::TemplateChild<gtk::Label>,
```

Maybe? Or how does this work exactly?

Currently this fails

```
error[E0412]: cannot find type `TemplateChild` in this scope
  --> src/lib.rs:40:20
   |
14 |         pub label: TemplateChild<gtk::Label>,
   |                    ^^^^^^^^^^^^^ not found in this scope
```

Reply:
The current code should work just fine. Even if I do as you suggest it fails to find super::MyWidget.

Reply:
Well, it doesn't find `glib` so `glib::wrapper` is not expanded and no `super::MyWidget` is defined :)

Reply:
Oh hm, but you do `use gtk::glib`. Then I have no idea what is wrong here. Seems like a bug

Reply:
It does sound like one, any ideas @GuillaumeGomez ?

Reply:
The example is run as a binary containing itself. `super` doesn't match anything in it.

Reply:
but the example is has a module `imp`, super is used inside that module. Which would default to everything outside the `imp` module. Unless i'm mis-understanding?